Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -54,14 +54,18 @@ final class TestDiscoveryCommand: CustomLLBuildCommand {
     ) throws {
         let stream = try LocalFileOutputByteStream(path)
 
+        let testsByClass = Dictionary(grouping: tests, by: { $0.name })
+
         stream <<< "import XCTest" <<< "\n"
         stream <<< "@testable import " <<< module <<< "\n"
 
-        for klass in tests {
+        for classTests in testsByClass {
```

Comment:
Since this is now iterating over a dictionary instead of an array, should the keys be sorted here to avoid spurious differences in how the output file is emitted?

Reply:
do you mean so that the output won't cause unnecessary compiles?

Reply:
Yes, that's what I meant — just to always have stable outputs.  Agreed that it can be done separately.

Reply:
https://github.com/apple/swift-package-manager/pull/3436